### PR TITLE
test(engine): add helm deployment

### LINKED ISSUE #6554: use helm to deploy dataflow engine in chaos test

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #6554

### What is changed and how it works?
 1. add helm deployment to dataflow engine
 2. use helm to deploy dataflow engine in chaos test
 3. fix some bugs in fakeJob

#### Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: engine/deployments/helm/tiflow/templates/metastore-etcd.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                fieldRef:
+                  fieldPath: metadata.name
+          ports:
+            - containerPort: 2479
+              name: port-etcd
+          command:
+            - "etcd"
+          args:
+            - "--name=$(MY_POD_NAME)"
+            - "--advertise-client-urls=http://0.0.0.0:2479"
```
Comment (gozssky) [inline engine/deployments/helm/tiflow/templates/metastore-etcd.yaml:65]: `advertise-client-urls` should use the service name.

File: Makefile [hunk 2]
```diff
@@ -500,6 +500,15 @@ engine_image:
 	@which docker || (echo "docker not found in ${PATH}"; exit 1)
 	./engine/test/utils/run_engine.sh build
 
+engine_image_linux: 
+	@which docker || (echo "docker not found in ${PATH}"; exit 1)
+	GOOS=linux GOARCH=amd64 $(GOBUILD) -ldflags '$(LDFLAGS)' -o bin/tiflow ./cmd/tiflow/main.go
+	docker build --platform linux/amd64 -f ./engine/deployments/docker/dev.Dockerfile -t dataflow:test ./
```
Comment (amyangfei) [inline Makefile:506]: Is it necessary to specific platform here

File: Makefile [hunk 3]
```diff
@@ -500,6 +500,15 @@ engine_image:
...
 	./engine/test/utils/run_engine.sh build
 
+engine_image_linux: 
+	@which docker || (echo "docker not found in ${PATH}"; exit 1)
+	GOOS=linux GOARCH=amd64 $(GOBUILD) -ldflags '$(LDFLAGS)' -o bin/tiflow ./cmd/tiflow/main.go
+	docker build --platform linux/amd64 -f ./engine/deployments/docker/dev.Dockerfile -t dataflow:test ./ 
+
+engine_image_infra: engine_image_linux
+	docker tag dataflow:test hub.pingcap.net/dataflow/engine:test
+	docker image push hub.pingcap.net/dataflow/engine:test
```
Comment (amyangfei) [inline Makefile:510]: It's an internal registry and needs user credential to login, let's remove it from Makefile.

File: engine/deployments/helm/tiflow/templates/metastore-etcd.yaml (hunk 1 above)
Comment (amyangfei) [reply engine/deployments/helm/tiflow/templates/metastore-etcd.yaml:65]: https://github.com/pingcap/tiflow/issues/6524#issuecomment-1200095965
/cc @CharlesCheung96

File: Makefile (hunk 2 above)
Comment (CharlesCheung96) [reply Makefile:506]: This speeds up compilation considerably, and makes it easy to compile images available to the test_infra platform on an ARM machine.

**Comment by CharlesCheung96** [comment]
Chaos test passed: https://github.com/CharlesCheung96/tiflow/actions/runs/2765902005

File: engine/deployments/helm/tiflow/templates/cases.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  template:
+    spec:
+{{- if .Values.cases.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.cases.imagePullSecrets | indent 6 }}
+{{- end }}
+      initContainers:
+        - name: wait-tiflow-cluster
+          image: busybox
+          command: ['sh', '-c', 'sleep 60']
```
Comment (amyangfei) [inline engine/deployments/helm/tiflow/templates/cases.yaml:17]: Will we use a health check API later?

File: engine/deployments/helm/tiflow/templates/executor.yaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: {{ include "tiflow.name" . }}-executor
+    spec:
+      initContainers:
+        - name: wait-server-master
+          image: busybox
+          command: ['sh', '-c', 'until nslookup {{ include "tiflow.name" . }}-server-master; do echo waiting for mysql; sleep 2; done;']
```
Comment (amyangfei) [inline engine/deployments/helm/tiflow/templates/executor.yaml:48]: ditto

File: engine/deployments/helm/tiflow/templates/server-master.yaml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    app.kubernetes.io/part-of: tiflow
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  ports:
+    - name: port-master
+      port: 10240
+      targetPort: 10240
+    - name: port-mpeer
+      port: 10239
+      targetPort: 10239
```
Comment (amyangfei) [inline engine/deployments/helm/tiflow/templates/server-master.yaml:20]: This is not needed.

File: engine/deployments/helm/tiflow/templates/cases.yaml (hunk 4 above)
Comment (CharlesCheung96) [reply engine/deployments/helm/tiflow/templates/cases.yaml:17]: Yes, we will add API check later.

Comment (amyangfei) [reply engine/deployments/helm/tiflow/templates/cases.yaml:17]: @gozssky prepares to add readiness API this week, can have a discussion about it.

**Review by amyangfei — APPROVED** [review]

File: engine/README.md [hunk 7]
```diff
@@ -113,4 +113,64 @@ cd ../../../ # root dir of tiflow
 ./bin/tiflow cli job create --master-addrs ${$server-master-ip0}:${port0} --job-type CVSDemo --job-config ./engine/deployments/docker-compose/config/demo.json
 ```
 
+## Manager engine cluster by [helm](https://github.com/helm/helm)
```
Comment (maxshuang) [inline engine/README.md:116]:
```suggestion
## Manager engine cluster by [helm](https://github.com/helm/helm) in local K8s
```

File: engine/pkg/rpcutil/client.go [hunk 8]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/pingcap/tiflow/pkg/retry"
 )
 
-const defaultDialRetry = 3
+const defaultDialRetry = 6
```
Comment (maxshuang) [inline engine/pkg/rpcutil/client.go:29]: Why we change this default value?

**Comment by maxshuang** [comment]
Greate job!!
After this pr, we can reuse the work in test-infra and deploy the tiflow cluster in local K8s for test.

**Review by maxshuang — APPROVED** [review]

File: engine/pkg/rpcutil/client.go (hunk 8 above)
Comment (CharlesCheung96) [reply engine/pkg/rpcutil/client.go:29]: This is only for test, i wiil fix it.

**Pull request merged**